neuralmonkey/encoders/cnn_encoder.py

```diff
@@ -299,15 +299,24 @@ def max_pooling(
             "Specification of a max-pooling layer (number {} in config) "
             'needs to have 3 members: "M", pool size, stride, padding, '
             "was {}").format(layer_num, specification))
-    pool_size, stride, pad = specification[1:]
+    pool_type, pool_size, stride, pad = specification
+
+    if pool_type == "M":
+        pool_fn = tf.layers.max_pooling2d
+    elif pool_type == "A":
+        pool_fn = tf.layers.average_pooling2d
+    else:
+        raise ValueError(
+            ("Unsupported type of pooling: {}, use 'M' for max-pooling or "
+             "'A' for average pooling.").format(pool_type))
 
     if pad not in ["same", "valid"]:
         raise ValueError(
             "Padding must be 'same' or 'valid', was '{}' in layer {}."
             .format(pad, layer_num + 1))
 
     with tf.variable_scope("layer_{}_max_pool".format(layer_num)):
-        next_layer = tf.layers.max_pooling2d(prev_layer, pool_size, stride)
+        next_layer = pool_fn(prev_layer, pool_size, stride)
         next_mask = tf.layers.max_pooling2d(prev_mask, pool_size, stride)
```

Comment:
tohle je furt to samý?

Reply:
Ne, když je to avereage, tak je to average, když je to max, tak je to max.

Reply:
takže to je tady  blbě... ?

Reply:
Jo, jestli myslíš ten řádek pod tím, tak je to pořád max: maska říká, které pixely jsou relevantní v nové kovoluční mapě a jsou ty,  ze kterých se bere nějaká hodnota z předchozí vrstvy, bez ohledu na to, že bere i z nějakého malého paddingu okolo, pokud je padding same. Kdyby tam byl average místo max, tak ta krajní maska bude mít nějaké floaty a ne 0/1.

Reply:
ok